src/jsonSchema.ts

```diff
@@ -0,0 +1,98 @@
+import { appendErrors, Resolver, transformToNestObject } from 'react-hook-form';
+//import * as Ajv from 'ajv';
+import Ajv, { ErrorObject, Options } from 'ajv';
+import type { JSONSchema4, JSONSchema6, JSONSchema7 } from 'json-schema';
+import {
+  FieldError,
+  FieldErrors,
+  ResolverError,
+  ResolverSuccess,
+} from 'react-hook-form/dist/types/form';
+
+export type JSONSchema = JSONSchema4 | JSONSchema6 | JSONSchema7;
```

Comment:
I am not family with json-schema, what's 4 6 7?

Reply:
Different standard versions. All of them are valid.

Reply:
I would mostly like to remove those type export in the final build as consistency with other resolvers, you can leave that to me tho.

Reply:
Uhm ok. `json-schema` is just a type definition package. I only wanted to add some form of type check to the schema like all the other implementations. The difference is that the schema is implied by the other libraries but ajv takes a schema with `any` type. It doesn't define the type of the schema. Instead it just throws if the schema is invalid. That's why I added the type.